nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/CheckBoxController.java

```diff
@@ -0,0 +1,145 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.LinearLayout;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Represents a field that allows a user to select multiple items from checkboxes.
+ * <p/>
+ * For the field value, the associated FormModel must return either a Set<String> or a Set<0-based index>, representing the
+ * currently selected items. Which representation to use is specified by the constructor. In either representation, no
+ * selection can be represented by returning {@code null} for the value of the field.
+ */
+public class CheckBoxController extends LabeledFieldController {
+    private final static int CONTAINER_ID = 1001;
+    private final static int CHECKBOX_ID = 101010;
+    private final List<String> items;
+    private final List<?> values;
+
+    /**
+     * Constructs a new instance of a checkboxes field.
+     *
+     * @param ctx              the Android context
+     * @param name             the name of the field
+     * @param labelText        the label to display beside the field. Set to {@code null} to not show a label
+     * @param isRequired       indicates if the field is required or not
+     * @param items            a list of Strings defining the selection items to show
+     * @param useItemsAsValues if true, {@code CheckBoxController} expects the associated form model to use
+     *                         the same string of the selected item when getting or setting the field; otherwise,
+     *                         {@code SelectionController} expects the form model to use index (as an Integer) to
+     *                         represent the selected item
+     */
+    public CheckBoxController(Context ctx, String name, String labelText, boolean isRequired, List<String> items, boolean useItemsAsValues) {
+        this(ctx, name, labelText, isRequired, items, useItemsAsValues ? items : null);
+    }
+
+    /**
+     * Constructs a new instance of a checkboxes field.
+     *
+     * @param ctx        the Android context
+     * @param name       the name of the field
+     * @param labelText  the label to display beside the field
+     * @param isRequired indicates if the field is required or not
+     * @param items      a list of Strings defining the selection items to show
+     * @param values     a list of Objects representing the values to set the form model on a selection (in
+     *                   the same order as the {@code items}.
+     */
+    public CheckBoxController(Context ctx, String name, String labelText, boolean isRequired, List<String> items, List<?> values) {
+        super(ctx, name, labelText, isRequired);
+        this.items = items;
+        this.values = values;
+    }
+
+    @Override
+    protected View createFieldView() {
+        LinearLayout checkboxContainer = new LinearLayout(getContext());
```

Comment:
Let's move the overall layout to an XML-based layout. The actual checkboxes can be inserted programmatically though. The main advantage is that allows the app to easily override the layout if needed.